flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterFactoryTest.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.slf4j;
+
+import org.apache.flink.metrics.reporter.MetricReporterFactory;
+import org.apache.flink.runtime.metrics.MetricReporterSetupTestBase;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Tests for loading the {@link Slf4jReporterFactory}.
+ */
+public class Slf4jReporterFactoryTest extends MetricReporterSetupTestBase {
+
+	@Override
+	protected List<Class<? extends MetricReporterFactory>> getMetricReporterFactoryClass() {
```

Comment:
redundant due to usage in e2e tests.

Reply:
I think e2e test does not focus on this feature and unit test should just focus on one thing. If we change the implementation of e2e test one day, who would guarantee the SPI work successfully?
I still prefer to include `slf4j` and `Prometheus` in the test scopes especially we already include tests for the rest 5 reporters.

Reply:
The Prometheus e2e test very much focuses on this, and tests all configurations one can think of (jar in plugins/lib, loaded via reflection/factory).
The reason it is necessary to test as an e2e test is that this feature working in production is reliant on the packaging; if the services stuff isn't included in the jar then it just won't work, and the unit test cannot catch that.

Reply:
After looking at the `PrometheusReporterEndToEndITCase`, I agree that it has already focused on this and I'll skip to cover on `slf4j` and `Prometheus` modules.